code/mapping/ext_modules/mapping_common/shape.py

```
@@ -0,0 +1,158 @@
+        m.scale.y = self.width
+
+        return m
+
+
+@dataclass(init=False)
+class Circle(Shape2D):
+    """Circle with radius in meters"""
+
+    radius: float
+
+    def __init__(self, radius: float, offset: Transform2D = Transform2D.identity()):
```

Ruff (0.8.2)

Do not perform function call `Transform2D.identity` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)